### PR TITLE
add LICENSE.txt to all packaged jars. (forward port to 8.0.x)

### PR DESCRIPTION
**Pull request by hrhino**
cf. #1783

Fixes scalaz/scalaz#1601 for real.

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2009-2017 Tony Morris, Runar Bjarnason, Tom Adams, Kristian Domagala, Brad Clow, Ricky Clarkson, Paul Chiusano, Trygve Laugstøl, Nick Partridge, Jason Zaugg, Julien Truffaut, Aloïs Cochard, Kenji Yoshida, Vincent Marquez, JB Giraudeau
+Copyright (c) 2009-2018 Tony Morris, Runar Bjarnason, Tom Adams, Kristian Domagala, Brad Clow, Ricky Clarkson, Paul Chiusano, Trygve Laugstøl, Nick Partridge, Jason Zaugg, Julien Truffaut, Aloïs Cochard, Kenji Yoshida, Vincent Marquez, JB Giraudeau
```
Comment (NeQuissimus) [inline LICENSE.txt]: Not to start anything but are these the people that we consider (and want to be named as) Scalaz 8 copyright holders?

File: project/ScalazBuild.scala [hunk 2]
```diff
@@ -35,23 +35,24 @@ object Scalaz {
...
+        compilerPlugin("org.spire-math"         %% "kind-projector"  % "0.9.6"),
+        compilerPlugin("com.github.tomasmikula" %% "pascal"          % "0.2.1"),
+        compilerPlugin("com.github.ghik"        %% "silencer-plugin" % "0.6")
+      ),
+      incOptions ~= (_.withLogRecompileOnMacro(false))
+    ) ++ {
+      Seq(packageBin, packageDoc, packageSrc).flatMap {
+        // include LICENSE.txt in all packaged artifacts
+        inTask(_)(Seq(mappings in Compile += licenseFile.value -> "LICENSE"))
+      }
```
Comment (hrhino) [inline project/ScalazBuild.scala]: dunno why this was formatted before and isn't now.

File: LICENSE.txt (hunk 1 above)
Comment (hrhino) [reply LICENSE.txt]: I have no opinion. There was discussion on #1783 about this re: the 7.x branches, but it went nowhere concrete.

File: .scalafmt.conf [hunk 3]
```diff
@@ -1,4 +1,4 @@
-maxColumn = 120
+maxColumn = 119
```
Comment (NeQuissimus) [inline .scalafmt.conf]: ...

Comment (hrhino) [reply .scalafmt.conf]: I don't see anything to see there.

File: LICENSE.txt (hunk 1 above)
Comment (hrhino) [reply LICENSE.txt]: For what it's worth, a quick analysis of `git blame` shows the following people as contributors to the `series/8.0.x` branch:

```bash
hhoughton:scalaz hhoughton$ git rev-list --format=%aN HEAD | grep -v '^commit' | sort -u | sed -e 's/^/- /'
```

- Alexander Konovalov
- Alois Cochard
- Brian McKenna
- Colin Woodbury
- Dale Wijnand
- Edmund Noble
- Emily Pillmore
- Filippo Mariotti
- Gaston Tonietti
- Harrison Houghton
- Igor Mielientiev
- Jean-Baptiste Giraudeau
- John A. De Goes
- Julien Truffaut
- Kamil Kloch
- Nic Flores
- Paweł Sączawa
- Stephen Compall
- Sukant Hajra
- Tim Steinbach
- Tomas Mikula
- Vincent
- Vincent Marquez
- kenji yoshida
- vincent
- wi101
- xuwei-k

**Comment by hrhino** [comment]
@fommil what think you about the list of authors in the `LICENSE` file?

**Comment by fommil** [comment]
We could opt in, but prompt everybody (occasionally)... maybe create a ticket and ask everybody who's contributed more than 50 lines or something like that.

**Comment by hrhino** [comment]
@fommil I suddenly wish we had a license like Apache with the `NOTICE` provision, so it could be generated by the build. I think people will forget to update the BSD file. 

I wonder if "copyright (c) 2018, the ScalaZ contributors" wouldn't just be best.

**Comment by fommil** [comment]
Nope, BSD is all about explicit lists :-P There would be no obligation to ship the contributor list with sources or binaries.

Of course, if everybody agrees to that one line change... we could do it. But I quite like that it's a big list of individuals not some kind of organisation.

Opt in means anybody can add themselves if they want. Some people prefer to be opt-out, e.g. I think @puffnfresh and @aloiscochard prefer to be cited but don't want to be part of any legal / conformist framework (I may misunderstand / misrepresent their view).

And who cares if it's a huge list... it's just a couple of bytes. My book will need to print it, but I'd probably format it to fit a single page to save the trees.

**Comment by hrhino** [comment]
> But I quite like that it's a big list of individuals not some kind of organi[z]ation.

I agree. "the ScalaZ contributors" was my attempt to make it sound as close to a group of individuals, not an organization (i.e., "copyright everyone who ever contributed to this).

> Opt in means anybody can add themselves if they want.

wait, really? I thought that that would require everyone to agree to any change, since it's technically "a change to the `LICENSE`". Or would we get blanket consent from everyone to later addition (but not removal) of names?

> And who cares if it's a huge list... it's just a couple of bytes. My book will need to print it, but I'd probably format it to fit a single page to save the trees.

Yeah, I couldn't care less about how many names go into it.

**Comment by fommil** [comment]
Technically we'd still need to contact previous copyright owners to add a single character... but at this point I'm assuming that everybody is fine with adding contributors who are pushing good code.

**Pull request merged**